knimin/lib/data_access.py

```diff
@@ -2399,64 +2400,205 @@ def getProjectNames(self):
         sql = """SELECT project FROM project"""
         return [x[0] for x in self._con.execute_fetchall(sql)]
 
+<<<<<<< HEAD
+    def get_id_by_name(self, field, name):
+        """Converts a field option name to its corresponding id
+
+        Parameters
+        ----------
+        field : str
+            Field name, i.e., table name under schema "pm"
+        name : str
+            Name of the option
+
+        Returns
+        -------
+        int
+            Id of the option
+
+        Raises
+        ------
+        ValueError
+            If name is not found in the table
+        """
+        sql = "SELECT {} FROM {} WHERE name = %s".format(field + '_id',
+                                                         'pm.' + field)
+        sql_args = [name]
+        id = self._con.execute_fetchone(sql, sql_args)
+        if id is None:
+            raise ValueError('"%s" is not a pre-defined option for %s.'
+                             % (name, field))
+        return id[0]
+
+    def create_sample(self, sample_ids):
+        """Creates samples by id
+
+        Parameters
+        ----------
+        sample_ids : list of str
+            Sample ids to create
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """INSERT INTO pm.sample (sample_id) VALUES (%s)"""
+        self._con.executemany(sql, [[x] for x in sample_ids])
+        return True
+
+    def delete_sample(self, sample_ids):
+        """Delete samples by id
+=======
     def get_plate_info(self, plate_id):
         """Gets attributes of a plate by id
+>>>>>>> 089a92844b668dcbfa9abd240295fc484482dc3d
 
         Parameters
         ----------
-        plate_id : int
-            Plate id
+        sample_ids : list of str
+            Sample ids to delete
 
         Returns
         -------
-        dict
-            Plate attributes
+        bool
+            True if successful
```

Comment:
They won't return False. They will only raise error if the SQL command doesn't run thru.